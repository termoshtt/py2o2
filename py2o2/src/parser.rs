//! Parse Python type stub file (*.pyi) generated by pyright
//!
//! The grammar of stub file is same as Python itself, descripted at
//! https://docs.python.org/3/library/ast.html#abstract-grammar

pub struct Stub {}

pub fn parse(pyi_input: &str) -> anyhow::Result<Stub> {
    todo!()
}
