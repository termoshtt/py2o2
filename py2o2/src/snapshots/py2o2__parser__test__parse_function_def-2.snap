---
source: py2o2/src/parser.rs
expression: "function_def(r#\"\n            def g(x: int) -> str:\n                ...\n            \"#.trim()).finish().unwrap()"
---
(
    "",
    FunctionDef {
        name: "g",
        args: [
            Arg {
                name: "x",
                ty: Some(
                    Name(
                        "int",
                    ),
                ),
                default: None,
            },
        ],
        type_: Name(
            "str",
        ),
    },
)
